pkg/providers/launchtemplate/launchtemplate.go

```diff
@@ -162,13 +164,28 @@ func (p *Provider) getStaticParameters(ctx context.Context, instanceType *cloudp
 		ClusterID:                      options.FromContext(ctx).ClusterID,
 		APIServerName:                  options.FromContext(ctx).GetAPIServerName(),
 		KubeletClientTLSBootstrapToken: options.FromContext(ctx).KubeletClientTLSBootstrapToken,
-		NetworkPlugin:                  options.FromContext(ctx).NetworkPlugin,
+		NetworkPlugin:                  getAgentbakerNetworkPlugin(ctx),
 		NetworkPolicy:                  options.FromContext(ctx).NetworkPolicy,
 		SubnetID:                       subnetID,
 		ClusterResourceGroup:           p.clusterResourceGroup,
 	}, nil
 }
 
+func getAgentbakerNetworkPlugin(ctx context.Context) string {
+	if isAzureCNIOverlay(ctx) || isCiliumNodeSubnet(ctx) {
+		return consts.NetworkPluginNone
+	}
+	return consts.NetworkPluginAzure
+}
+
+func isCiliumNodeSubnet(ctx context.Context) bool {
+	return options.FromContext(ctx).NetworkPlugin == consts.NetworkPluginAzure && options.FromContext(ctx).NetworkPluginMode == consts.NetworkPluginModeNone && options.FromContext(ctx).NetworkDataplane == consts.NetworkDataplaneCilium
+}
+
+func isAzureCNIOverlay(ctx context.Context) bool {
+	return options.FromContext(ctx).NetworkPlugin == consts.NetworkPluginAzure && options.FromContext(ctx).NetworkPluginMode == consts.NetworkPluginModeOverlay
+}
```

Comment:
Should be extracting settings from context and passing into helpers, rather than making them depend on context. Generally, the earlier settings are extracted from context the better.